source/hash_table.c

```diff
@@ -653,38 +671,60 @@ static inline void s_get_next_element(struct aws_hash_iter *iter, size_t start_s
         if (entry->hash_code) {
             iter->element = entry->element;
             iter->slot = i;
+            iter->status = AWS_HASH_ITER_STATUS_READY_FOR_USE;
             return;
         }
     }
     iter->element.key = NULL;
     iter->element.value = NULL;
     iter->slot = iter->limit;
+    iter->status = AWS_HASH_ITER_STATUS_DONE;
```

Comment:
Is this a bug fix? Please mention in the commit message if so

Reply:
See above.